gradebookng/tool/src/webapp/styles/gradebook-grades.css

```diff
@@ -565,3 +565,125 @@ div.wicket-modal div.w_right > div {
   margin: 0;
 }
 
+.gb-noGbItems
+{
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  width: 100%;
+  padding: 1em;
+  font-size: 20px;
+  text-align: center;
+  color: #666;
+  background-color: #eee;
+  -ms-grid-row: 3;
+  -ms-grid-row-span: 1;
+  -ms-grid-column: 2;
+  -ms-grid-column-span: 1;
+  grid-area: empty;
+}
+.gb-noGbItems > i
+{
+  display: block;
+  font-size: 3em;
+}
+
+.gradeTableOuterWrapper
+{
+  -ms-grid-row: 3;
+  -ms-grid-row-span: 1;
+  -ms-grid-column: 1;
+  -ms-grid-column-span: 1;
+  grid-area: grades;
+}
+
+.gradesToolbar1
+{
+  -ms-grid-row: 1;
+  -ms-grid-row-span: 1;
+  -ms-grid-column: 1;
+  -ms-grid-column-span: 2;
+  grid-area: toolbar1;
+}
+
+.gradesToolbar2
+{
+  -ms-grid-row: 2;
+  -ms-grid-row-span: 1;
+  -ms-grid-column: 1;
+  -ms-grid-column-span: 2;
+  grid-area: toolbar2;
+}
+
+.gradeTableArea
+{
+  display: -ms-grid;
+  display: grid;
+  -ms-grid-columns: auto 1fr;
+  grid-template-columns: auto 1fr;
+  -ms-grid-rows: auto auto auto;
+  grid-template-areas:
+    "toolbar1 toolbar1"
+    "toolbar2 toolbar2"
+    "grades empty";
+}
```

Comment:
So does this officially kill our implied support of IE 11? I know IE11 isn't our best-supported browser, but I thought it was decent enough to be usable.... Do we know what happens in IE 11 with this grid template?

Reply:
No, it actually works reasonably well in IE11, that's what the -ms-grid-* properties are for. IE11 supports the original spec for Grid so I was able in this case to use the IE11 equivalent syntax to achieve the same goal.

Reply:
That said, IE11 is really starting to hold us back with the release of Grid support in all modern browsers, and its lack of HTML5 support in general.

Reply:
Okay, good news re -ms-grid.

It looks like Jan 2020 for Win 7 extended EOL .... So we should start deprecating IE11 in next release.